plugin/cmd/kube-scheduler/scheduler.go

```diff
@@ -36,5 +37,8 @@ func main() {
 
 	verflag.PrintAndExitIfRequested()
 
-	app.Run(s)
+	if err := app.Run(s); err != nil {
+		glog.Errorf("scheduler app failed to run: %v", err)
+	}
+	glog.Fatal("this statement shouldn't be reached")
```

Comment:
we should panic on a unreachable line not fatal. usually fatal on unrecoverable external error, and panic on unrecoverable internal error.